pkg/resourcequota/resourcequota.go

```diff
@@ -0,0 +1,113 @@
+package resourcequota
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/julienschmidt/httprouter"
+	"github.com/opencost/opencost/core/pkg/log"
+	proto "github.com/opencost/opencost/core/pkg/protocol"
+
+	"github.com/opencost/opencost/pkg/env"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+var protocol = proto.HTTP()
+
+type QuotaTracker struct {
+	clientset *kubernetes.Clientset
+}
+
+type QuotaMetrics struct {
+	ClusterID    string            `json:"clusterID"`
+	Namespace    string            `json:"namespace"`
+	QuotaName    string            `json:"quotaName"`
+	ResourceType string            `json:"resourceType"`
+	Hard         resource.Quantity `json:"hard"`
+	Used         resource.Quantity `json:"used"`
+	Percentage   float64           `json:"percentage"`
+	Timestame    time.Time         `json:"timestamp"`
+	Labels       map[string]string `json:"labels"`
+	Recommended  resource.Quantity `json:"recommended"`
+}
+
+func NewQuotaTracker(clientset *kubernetes.Clientset) *QuotaTracker {
+	return &QuotaTracker{
+		clientset: clientset,
+	}
+}
+
+func (qt *QuotaTracker) GetAllQuotaMetrics() (map[string]map[string][]QuotaMetrics, error) {
+	allMetrics := make(map[string]map[string][]QuotaMetrics)
+	namespaces, err := qt.clientset.CoreV1().Namespaces().List(context.Background(), metav1.ListOptions{})
```

Comment:
This will change to `GetAllNamespaces()` when you substitute the clientset for `clustercache.ClusterCache`